templates/outer-loop/gradle/Dockerfile

```diff
@@ -0,0 +1,88 @@
+# Step 1: Build the user's application.
+FROM {{.STACK_IMAGE_GRADLE}} as compile
+
+# Make a well known place for shared library jars separate from the rest of the <server> contents (to help with caching).
+RUN mkdir /work/configlibdir \
+   && mkdir /work/config \
+   &&  mkdir /work/shared
+
+# Copy application source and build data.
+COPY --chown=1001:0 ./. /work/outer-loop-app/tmp/
+RUN cp -r /work/outer-loop-app/tmp/src /work/outer-loop-app/.; \
```

Comment:
IIUC, the reason this looks so complicated, with the gradle globbing, is that we want to avoid copying someones target/build dir into the docker build container.    Should we just do a regex copy or find piped into a copy somehow or some other tool to copy "everything but the build dir"? 

The same comment applies to maven, where we copy /src and pom.xml but not say /data that possibly someone might want.

Reply:
I'm more concerned with the "contract" than the caching.    For mvn it is easy to understand that we COPY in /src and /pom.xml.   What's our rationale for including these gradle-related globs?  What parts are we gathering?

Note we could choose to say:  "copy everything and you should be using .dockerignore to filter it out".  Maybe our starter and least should have a .dockerignore.

Reply:
Opened https://github.com/OpenLiberty/application-stack/issues/156